pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -1063,14 +1066,9 @@ func newAWSCloud(cfg CloudConfig, awsServices Services) (*Cloud, error) {
 		return nil, err
 	}
 
-	// Trust that if we get a region from configuration or AWS metadata that it is valid,
```

Comment:
Good point, I've added that back. The previous behaviour wasn't quite correct either since `updateConfigZone` allows the user to override the zone and the existing code will recognize it even if that's not truly a valid region. The credential provider behaviour seems to still be correct.